gateway/ln-gateway/src/lib.rs

```diff
@@ -840,6 +842,18 @@ impl Gateway {
         Err(GatewayError::Disconnected)
     }
 
+    async fn handle_leave_federation(&mut self, payload: LeaveFedPayload) -> Result<()> {
+        self.remove_client(payload.federation_id).await?;
+        let mut dbtx = self.gateway_db.begin_transaction().await;
+        dbtx.remove_entry(&FederationIdKey {
+            id: payload.federation_id,
+        })
+        .await;
+        dbtx.commit_tx_result()
+            .await
+            .map_err(GatewayError::DatabaseError)
```

Comment:
We explicitly don't delete the client DB here since that could lead to loss of funds. This means joining the same federation again will get you all the old state back. If that's not what we want we could just rename the DB folder into something like `<fed-id>.bak[.<index>]`.

Reply:
> We explicitly don't delete the client DB here since that could lead to loss of funds. This means joining the same federation again will get you all the old state back.

Makes sense to me

> ... we could just rename the DB folder into something like `<fed-id>.bak[.<index>]`.

Seems like this could be under backup/restore apis, and `leave-federation` would do a backup